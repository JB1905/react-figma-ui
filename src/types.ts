import { colorNames, iconNames, sizes, tints, weights } from './constants';

export type ColorName = (typeof colorNames)[number];
export type IconName = (typeof iconNames)[number];
export type Size = (typeof sizes)[number];
export type Tint = (typeof tints)[number];
export type Weight = (typeof weights)[number];
