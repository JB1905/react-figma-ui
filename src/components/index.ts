export * from './Button';
export * from './Checkbox';
export * from './Disclosure';
export * from './Icon';
export * from './IconButton';
export * from './Input';
export * from './Label';
export * from './Onboarding';
export * from './Radio';
export * from './SectionTitle';
export * from './SelectMenu';
export * from './Switch';
export * from './Textarea';
export * from './Type';
