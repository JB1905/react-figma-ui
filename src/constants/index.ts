export * from './colorNames';
export * from './iconNames';
export * from './sizes';
export * from './tints';
export * from './weights';
