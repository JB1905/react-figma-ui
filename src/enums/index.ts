export { Size } from './Size';
export { Tint } from './Tint';
export { Weight } from './Weight';
