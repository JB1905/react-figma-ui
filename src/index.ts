export { Button } from './components/Button';
export { Checkbox } from './components/Checkbox';
export { Disclosure } from './components/Disclosure';
export { Icon } from './components/Icon';
export { IconButton } from './components/IconButton';
export { Input } from './components/Input';
export { Label } from './components/Label';
export { Onboarding } from './components/Onboarding';
export { Radio } from './components/Radio';
export { SectionTitle } from './components/SectionTitle';
export { SelectMenu } from './components/SelectMenu';
export { Switch } from './components/Switch';
export { Textarea } from './components/Textarea';
export { Type } from './components/Type';
