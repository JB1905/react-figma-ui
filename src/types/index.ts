import { colorNames } from '../constants/colorNames';
import { iconNames } from '../constants/iconNames';
import { sizes } from '../constants/sizes';
import { tints } from '../constants/tints';
import { weights } from '../constants/weights';

export type ColorName = typeof colorNames;
export type IconName = typeof iconNames;
export type Size = typeof sizes;
export type Tint = typeof tints;
export type Weight = typeof weights;
