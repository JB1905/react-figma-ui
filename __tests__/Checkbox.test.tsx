import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import { Checkbox } from '../src';

describe('Checkbox', () => {
  // TODO: check, uncheck
  // TODO onChange

  it('should render unchecked Checkbox', () => {
    const onClick = jest.fn();

    const { container, getByText } = render(
      <Checkbox id="uniqueId" readOnly onClick={onClick}>
        Label
      </Checkbox>
    );

    expect(container).toMatchSnapshot();

    fireEvent.click(getByText('Label'));

    // expect(onClick).not.toHaveBeenCalled();
    expect(onClick).toHaveBeenCalled();

    // TODO: onClick to have been called
    // TODO: not to be checked
  });

  it('should render checked Checkbox', () => {
    const onClick = jest.fn();

    const { container } = render(
      <Checkbox id="uniqueId" readOnly checked onClick={onClick}>
        Label
      </Checkbox>
    );

    expect(container).toMatchSnapshot();

    // fireEvent.click(getByText('Label'));

    // expect(onClick).not.toHaveBeenCalled();

    // TODO: onClick not to have been called
    // TODO: to be checked
  });

  it('should render disabled Checkbox', () => {
    const onClick = jest.fn();

    const { container, getByText } = render(
      <Checkbox id="uniqueId" readOnly disabled>
        Label
      </Checkbox>
    );

    expect(container).toMatchSnapshot();

    fireEvent.click(getByText('Label'));

    expect(onClick).not.toHaveBeenCalled();
  });

  it('should render Checkbox with custom className values', () => {
    const onClick = jest.fn();

    const { container, getByText } = render(
      <Checkbox
        id="uniqueId"
        className="custom-class-name"
        containerProps={{ className: 'custom-class-name' }}
        labelProps={{ className: 'custom-class-name' }}
        onClick={onClick}
      >
        Label
      </Checkbox>
    );

    expect(container).toMatchSnapshot();

    fireEvent.click(getByText('Label'));

    expect(onClick).toHaveBeenCalled();
  });
});
