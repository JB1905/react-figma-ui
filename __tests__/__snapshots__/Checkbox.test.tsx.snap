// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render Checkbox with custom className values 1`] = `
<div>
  <div
    class="checkbox custom-class-name"
  >
    <input
      class="checkbox__box custom-class-name"
      id="uniqueId"
      type="checkbox"
    />
    <label
      class="checkbox__label custom-class-name"
      for="uniqueId"
    >
      Label
    </label>
  </div>
</div>
`;

exports[`Checkbox should render checked Checkbox 1`] = `
<div>
  <div
    class="checkbox"
  >
    <input
      checked=""
      class="checkbox__box"
      id="uniqueId"
      readonly=""
      type="checkbox"
    />
    <label
      class="checkbox__label"
      for="uniqueId"
    >
      Label
    </label>
  </div>
</div>
`;

exports[`Checkbox should render disabled Checkbox 1`] = `
<div>
  <div
    class="checkbox"
  >
    <input
      class="checkbox__box"
      disabled=""
      id="uniqueId"
      readonly=""
      type="checkbox"
    />
    <label
      class="checkbox__label"
      for="uniqueId"
    >
      Label
    </label>
  </div>
</div>
`;

exports[`Checkbox should render unchecked Checkbox 1`] = `
<div>
  <div
    class="checkbox"
  >
    <input
      class="checkbox__box"
      id="uniqueId"
      readonly=""
      type="checkbox"
    />
    <label
      class="checkbox__label"
      for="uniqueId"
    >
      Label
    </label>
  </div>
</div>
`;
